protocols/imap/src/main/java/org/apache/james/imap/decode/parser/SetACLCommandParser.java

```diff
@@ -43,10 +48,29 @@ public SetACLCommandParser(StatusResponseFactory statusResponseFactory) {
 
     @Override
     protected ImapMessage decode(ImapRequestLineReader request, Tag tag, ImapSession session) throws DecodingException {
-        final String mailboxName = request.mailbox();
-        final String identifier = request.astring();
-        final String rights = request.astring();
-        request.eol();
-        return new SetACLRequest(tag, mailboxName, identifier, rights);
+        try {
+            final SetACLRequest.MailboxName mailboxName = new SetACLRequest.MailboxName(request.mailbox());
+            final Username identifier = Username.of(request.astring());
+            final String editModeAndRights = request.astring();
+            request.eol();
+
+            MailboxACL.EditMode editMode = MailboxACL.EditMode.REPLACE;
+            if (StringUtils.isNotEmpty(editModeAndRights)) {
+                switch (editModeAndRights.charAt(0)) {
+                    case MailboxACL.ADD_RIGHTS_MARKER:
+                        editMode = MailboxACL.EditMode.ADD;
+                        break;
+                    case MailboxACL.REMOVE_RIGHTS_MARKER:
+                        editMode = MailboxACL.EditMode.REMOVE;
+                        break;
+                }
+            }
+
+            MailboxACL.Rfc4314Rights rights = MailboxACL.Rfc4314Rights.deserialize(editModeAndRights.substring(1));
+
+            return new SetACLRequest(tag, mailboxName, identifier, editMode, rights);
```

Comment:
That's... brutal! And a lot of parameter.

Having a high count of method parameter is a code smell.

MailboxACL actually is a nice POJO to bundle ` identifier, editMode, rights` together. Can't we use it?